cmd/podman/images/scan.go

```diff
@@ -0,0 +1,156 @@
+package images
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/podman/v2/cmd/podman/common"
+	"github.com/containers/podman/v2/cmd/podman/registry"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/containers/podman/v2/pkg/specgen"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+var (
+	scanDescription = `Scan a container image for known vulnerabilities.  The image name or digest can be used.`
+	scanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman scan centos:latest`,
+	}
+
+	imageScanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    scanDescription,
+		RunE:    scan,
+		Example: `podman image scan centos:latest`,
+	}
+)
+
+var (
+	scanOptions = entities.ImageScanOptions{
+		ContainerRunOptions: entities.ContainerRunOptions{
+			OutputStream: os.Stdout,
+			ErrorStream:  os.Stderr,
+			Rm: true,
+		},
+	}
+
+	cliVals common.ContainerCLIOpts
+)
+
+func scanFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	common.DefineCreateFlags(cmd, &cliVals)
+	common.DefineNetFlags(cmd)
+
+	// TODO: derive default from containers.conf configuration
+	scannerFlagName := "scanner"
+	flags.StringVar(&scanOptions.ScannerImage, scannerFlagName, "docker.io/library/grype:latest", "The vulnerability scanner container image to use")
+}
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: scanCommand,
+	})
+	scanFlags(scanCommand)
+
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: imageScanCommand,
+		Parent:  imageCmd,
+	})
+	scanFlags(imageScanCommand)
+
+}
+
+func scan(cmd *cobra.Command, args []string) error {
+	var err error
+
+	if len(args) < 1 {
```

Comment:
Other commands use Cobra's arg checks to make sure we have a consistent error reporting.  There's an `Args` field in the `cobra.Command` struct that we can set to `cobra.ExactArgs(1)`.

Reply:
I was thinking about swapping this to `MinimumNArgs` to keep allowing for arguments to be passed to the run command (the scanner). Do you think we should allow this here in the CLI? or only allow extra options through configuration?

Reply:
I suggest to start with exactly one arg for now. If we need more in the future, we can change :)

Reply:
I feel that since the scan tool will be agnostic that the user will want to pass options to change behavior (output verbosity, format options, scanner profiles/policies, etc.). The CLI is a natural way to achieve this:

```
# where anything after "--" is passed as arguments to the scanner container
podman image scan myimage:latest -- -o json -v --fail-on critical
```

That being said, I'm open for other options how to achieve this... maybe this can be replaced with `container.conf` options (or similar)? Where do you think these options should go in the context of podman configuration?